From e421c890d68467bcd6ef067c0bdf3b2953e950e4 Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Wed, 12 Feb 2020 21:33:15 +0300
Subject: [PATCH] Disable LZMA support

KArchive links to LZMA incorrectly on Windows. It links to liblzma-5.dll,
even though we have only liblzma.dll. Since we don't use .xz functionality
of KArchive, we can just disable it on Windows.

NOTE: a proper solution would be to play with cmake's LINK_INTERFACE
flags and explain that we should link to this .dll without version
suffix.
---
 CMakeLists.txt | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f21657..d0460c1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,13 +36,16 @@ set_package_properties(BZip2 PROPERTIES
     PURPOSE "Support for BZip2 compressed files and data streams"
 )
 
-find_package(LibLZMA)
-set_package_properties(LibLZMA PROPERTIES
-    URL "http://tukaani.org/xz/"
-    DESCRIPTION "Support for xz compressed files and data streams"
-    TYPE OPTIONAL
-    PURPOSE "Support for xz compressed files and data streams"
-)
+if (NOT WIN32)
+   find_package(LibLZMA)
+   set_package_properties(LibLZMA PROPERTIES
+                          URL "http://tukaani.org/xz/"
+                          DESCRIPTION "Support for xz compressed files and data streams"
+                          TYPE OPTIONAL
+                          PURPOSE "Support for xz compressed files and data streams"
+                          )
+endif()
+
 include_directories(
     ${ZLIB_INCLUDE_DIR}
 )
-- 
2.20.1.windows.1

